tests/test_generated.py

```diff
@@ -0,0 +1,81 @@
+from google.api_core.exceptions import BadRequest, NotFound
+from google.cloud import bigquery
+from .util import coerce_result, generate_tests
+
+import json
+import os
+import pytest
+
+
+@pytest.fixture(scope="session")
+def bq():
+    return bigquery.Client(project=os.environ.get("GCP_PROJECT", None))
```

Comment:
there's probably a better named env var to use here

Reply:
Is there no python equivalent to the [Java SDK's testing helpers](https://github.com/googleapis/google-cloud-java/blob/master/TESTING.md#testing-code-that-uses-bigquery)? With the Java machinery, it's able to read creds from env vars and you can then extract the default project id based on the creds.

Reply:
Due to the priority of various things there's no built in env var that can override the gcloud cli default project but still use the gcloud default creds. default creds and not setting `GCP_PROJECT` also works fine. I'll extract this to a later PR for now.